lib/gollum/app.rb

```diff
@@ -179,7 +179,7 @@ def wiki_new
       halt 500 unless tempfile.is_a? Tempfile
 
       # Remove page file dir prefix from upload path if necessary -- committer handles this itself
-      dir      = wiki.per_page_uploads ? params[:upload_dest].match(/^(#{wiki.page_file_dir}\/+)?(.*)/)[2] : 'uploads'
+      dir      = wiki.per_page_uploads ? params[:upload_dest] : ::File.join([wiki.page_file_dir, 'uploads'].compact)
```

Comment:
that's precisely what I had in mind. This change seems inevitable, so there is not much sense in doing 'just-for-now' security checks.